model/websocket_client.go

```diff
@@ -29,7 +29,13 @@ type WebSocketClient struct {
 // NewWebSocketClient constructs a new WebSocket client with convenience
 // methods for talking to the server.
 func NewWebSocketClient(url, authToken string) (*WebSocketClient, *AppError) {
-	conn, _, err := websocket.DefaultDialer.Dial(url+API_URL_SUFFIX_V3+"/users/websocket", nil)
+	return NewWebSocketClientWithDialer(url, authToken, websocket.DefaultDialer)
+}
+
+// NewWebSocketClientWithDialer constructs a new WebSocket client with convienence
+// methods for talking to the server using a custom dialer.
+func NewWebSocketClientWithDialer(url, authToken string, dialer *websocket.Dialer) (*WebSocketClient, *AppError) {
```

Comment:
Can you make `dialer` the first parameter here (and below), mimicking the pattern set by https://golang.org/pkg/crypto/tls/#DialWithDialer?